### ISSUE TITLE
Motion Tigre - Error using mex

### ISSUE THREAD

**Issue by aneelmaheshwari** [AnderBiguri/MotionTIGRE#2]
i am using the Tigre but i am new with motion Tigre. Do we have to put the motion Tigre folder in the main Tigre folder or separate. I have checked all the Mex files in the correct folder for Motion Tigre but still there is a MeX error in motion Tigre. Therefore, I wouldn't be able to compile the program.

Error using mex
C:\TIGRE-master\TIGRE-master\MATLAB\Motion_Source\AxM.cpp not detected; check that you are in the correct current folder, and
check the spelling of 'C:\TIGRE-master\TIGRE-master\MATLAB\Motion_Source\AxM.cpp'.

Someone helps in this regard.

**Comment by AnderBiguri** [comment]
Please, can you post the actual error message, so I can help?

**Comment by AnderBiguri** [comment]
Have you followed the demo? It has instructions
https://github.com/AnderBiguri/MotionTIGRE/blob/master/MotionDemo.m

If you have, can you let me know what of this didn't work? By your error message it just looks like you tried to compile in a folder that didn't have the files.

**Comment by aneelmaheshwari** [comment]
In Tigre, there lot step has to follow to complete so it can work. but in Motion Tigre, i hav download and save in C drive. open the folder and run the MotionDemo.m. I was trying to run the code but it providing the above 
C:\TIGRE-master\TIGRE-master\MATLAB\Motion_Source\AxM.cpp not detected. it keep giving same mex error.

**Comment by aneelmaheshwari** [comment]
Error in MotionDemo (line 33)
mex -largeArrayDims ./Motion_Source/AxM.cpp   ./Motion_Source/ray_interpolated_projection_motion.cu ./Motion_Source/Siddon_projection.cu -outdir ./Motion_Correction

**Comment by AnderBiguri** [comment]
Right, but is `./Motion_Source/AxM.cpp` there from the folder you are in (`./` means a subfolder of the current location)? Because the first error you put suggest the file is not there. Please do post the full error messages, this last one only says there is an error. It looks like you simply are not in the right folder to compile

**Comment by aneelmaheshwari** [comment]
but file in the subfolder but still giving error 
Error using mex

C:\TIGRE-master\TIGRE-master\MATLAB\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\Siddon_projection.cu not detected; check
that you are in the correct current folder, and check the spelling of
'C:\TIGRE-master\TIGRE-master\MATLAB\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\Siddon_projection.cu'.

Error in MotionDemo (line 33)
mex -largeArrayDims Motion_Source/AxM.cpp   ./Motion_Source/ray_interpolated_projection_motion.cu ./Motion_Source/Siddon_projection.cu -outdir ./Motion_Correction

**Comment by aneelmaheshwari** [comment]
Siddon_projection.cu ? this is not in the folder

**Comment by AnderBiguri** [comment]
@aneelmaheshwari  ah, I see. Not sure, this is a file from TIGRE, try to just copy paste it from there. TIGRE has changed a lot since this code, so its not guaranteed to work.

**Comment by aneelmaheshwari** [comment]
but i can't find this Siddon_projection.cu  in Tigre?I have checked in the mex file, but there not Siddon_projection.cu

**Comment by AnderBiguri** [comment]
https://github.com/CERN/TIGRE/tree/master/Common/CUDA

**Comment by aneelmaheshwari** [comment]
Now i am getting this error

Building with 'NVIDIA CUDA Compiler'.
Error using mex
nvcc warning : The 'compute_35', 'compute_37', 'compute_50', 'sm_35', 'sm_37' and 'sm_50' architectures are deprecated, and may be
removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
AxM.cpp
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\Siddon_projection.hpp(48): fatal error C1083: Cannot open include file:
'ray_interpolated_projection.hpp': No such file or directory



Error in MotionDemo (line 33)
mex -largeArrayDims ./Motion_Source/AxM.cpp   ./Motion_Source/ray_interpolated_projection_motion.cu ./Motion_Source/Siddon_projection.cu -outdir ./Motion_Correction
 
>>

**Comment by AnderBiguri** [comment]
Please, do read the errors before posting them here. As you can read from the error, it is exactly the same one as you got in the first time, except now it refers to another file. Therefore, the solution is exactly the same as the previous solution, except you now need to use a different file. 

Unfortunately, as I said in my email, MotionTIGRE is essentially abandoned and not supported. I would be happy to give a slight guidance from time to time if it doesn't take much time, but in theory, this code is not officially supported and therefore I can't spend my time answering each error that it comes with, sadly.

**Comment by aneelmaheshwari** [comment]
Please just check this final time and i have copy all files which came in error

Error using mex
nvcc warning : The 'compute_35', 'compute_37', 'compute_50', 'sm_35', 'sm_37' and 'sm_50' architectures are deprecated, and may be
removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
AxM.cpp
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\AxM.cpp(422): error C2660: 'siddon_ray_projection': function does not take
5 arguments
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\Siddon_projection.hpp(54): note: see declaration of 'siddon_ray_projection'
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\AxM.cpp(424): warning C4267: 'argument': conversion from 'size_t' to 'int',
possible loss of data
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\AxM.cpp(445): warning C4018: '<': signed/unsigned mismatch



Error in MotionDemo (line 33)
mex -largeArrayDims ./Motion_Source/AxM.cpp   ./Motion_Source/ray_interpolated_projection_motion.cu ./Motion_Source/Siddon_projection.cu -outdir ./Motion_Correction
 
>>

**Comment by aneelmaheshwari** [comment]
Please  just guide fianl time on this if possible 

Error in MotionDemo (line 33)
mex -largeArrayDims ./Motion_Source/AxM.cpp ./Motion_Source/ray_interpolated_projection_motion.cu ./Motion_Source/Siddon_projection.cu -outdir ./Motion_Correction

**Comment by AnderBiguri** [comment]
Unfortunately the answer to the error (the one you deleted, the current line does not say what error it is, I insist you need to put the actual error message) is that the current TIGRE is not compatible with this code, only the 2018 TIGRE is, and at some point things changed. You can either use the 2018 version (you can find this using github history of TIGRE), or you need to learn how it works and fix it yourself. Its not a complex fix, but it is one that requires you to learn how to tackle all these minor issues yourself, definitely minimally reading error messages. 
Unfortunately I can't help much more.

**Comment by aneelmaheshwari** [comment]
Error using mex
nvcc warning : The 'compute_35', 'compute_37', 'compute_50', 'sm_35', 'sm_37' and 'sm_50' architectures are deprecated, and may be
removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
AxM.cpp
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\AxM.cpp(422): error C2660: 'siddon_ray_projection': function does not take
5 arguments
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\Siddon_projection.hpp(54): note: see declaration of 'siddon_ray_projection'
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\AxM.cpp(424): warning C4267: 'argument': conversion from 'size_t' to 'int',
possible loss of data
C:\MotionTIGRE-master\MotionTIGRE-master\Motion_Source\AxM.cpp(445): warning C4018: '<': signed/unsigned mismatch



Error in MotionDemo (line 33)
mex -largeArrayDims ./Motion_Source/AxM.cpp   ./Motion_Source/ray_interpolated_projection_motion.cu ./Motion_Source/Siddon_projection.cu -outdir ./Motion_Correction
 
>>

**Comment by aneelmaheshwari** [comment]
I respect and appreciate your time. Thanks
